<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Introduction to protrusions proteome analysis • protrusionproteome</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Introduction to protrusions proteome analysis">
<meta property="og:description" content="protrusionproteome">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">protrusionproteome</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.99.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/protrusionproteome_BC.html">Introduction to protrusions proteome analysis</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="protrusionproteome_BC_files/header-attrs-2.6/header-attrs.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Introduction to protrusions proteome analysis</h1>
                        <h4 class="author">Maria Dermit</h4>
            
            <h4 class="date">16 January 2021</h4>
      
      
      <div class="hidden name"><code>protrusionproteome_BC.Rmd</code></div>

    </div>

    
    
<div style="page-break-after: always;"></div>
<div id="abstract" class="section level1">
<h1 class="hasAnchor">
<a href="#abstract" class="anchor"></a>Abstract</h1>
<p><code>protrusionproteome</code> is a package that provides an analytical workflow of shotgunmass spectrometry-based proteomics experiments with tandem mass tag (TMT) labelingof protrusion profiling experiments <span class="citation">(Maria Dermit 2020)</span>. Protrusion are purified from microporous transwell filters and their proteomes are compare to the cell bodies.</p>
<p>protrusionproteome requires tabular input e.g. proteinGroups.txt, peptides.txt and evidence.txt output of quantitative analysis software like MaxQuant. Functions are provided for preparation, filtering as well as log transformation,calculating TMT ratios and median substation as well as generating SummarizedExperiment objects. It also includes tools to visualize protrusion purification efficiency, protease miscleavages and TMT incorporation efficiency with visualization such as scatterplot and barplot representations. Finally, it includes statistical testing of significantly enrich categories in cell protrusions.</p>
</div>
<div id="installation-and-loading-package" class="section level1">
<h1 class="hasAnchor">
<a href="#installation-and-loading-package" class="anchor"></a>Installation and loading package</h1>
<p>Start R and install the RBP package:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html">requireNamespace</a></span><span class="op">(</span><span class="st">"BiocManager"</span>, quietly<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span>
    <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span>
<span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/install.html">install</a></span><span class="op">(</span><span class="st">"protrusionproteome"</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="st">"protrusionproteome"</span><span class="op">)</span></code></pre></div>
<p>Once you have the package installed, load <code>protrusionproteome</code> and dplyr for data transformation into R.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">protrusionproteome</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org">dplyr</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">SummarizedExperiment</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="http://lubridate.tidyverse.org">lubridate</a></span><span class="op">)</span></code></pre></div>
</div>
<div id="getting-the-data" class="section level1">
<h1 class="hasAnchor">
<a href="#getting-the-data" class="anchor"></a>Getting the data</h1>
<p>We analyze the time-course protrusion profiles proteome dataset from <span class="citation">(Maria Dermit 2020)</span> (PRIDE PXD021239).</p>
<p>The raw mass spectrometry data were first analyzed using MaxQuant <span class="citation">(Jürgen Cox 2014)</span>. We initially perform a number of data cuality checks including trypsin efficiency and TMT labeling incorporation.</p>
<div id="qc-calculating-protease-efficiency" class="section level2">
<h2 class="hasAnchor">
<a href="#qc-calculating-protease-efficiency" class="anchor"></a>QC: Calculating protease efficiency</h2>
<p>Tryspsin was the protease used to produce tryptic peptides for this dataset. Note that the maximum missed cleavages allowed on the MaxQuant search was 2. The numbers of trypsin miscleavages can be used as a proxy of trypsin efficiency. This information is contained in the “peptides.txt” file, and is provided with this package:</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Peptides data is provided with the package</span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="st">"peptides.raw"</span><span class="op">)</span>
<span class="va">pepdata</span> <span class="op">&lt;-</span> <span class="va">peptides.raw</span></code></pre></div>
<p>To visualize trypsin efficiency, the plot_miscleavagerate function can be used.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Stacked barplot of trypsin miscleavages</span>
<span class="fu"><a href="../reference/plot_miscleavagerate.html">plot_miscleavagerate</a></span><span class="op">(</span><span class="va">pepdata</span><span class="op">)</span></code></pre></div>
<p><img src="protrusionproteome_BC_files/figure-html/plot-miscleavages-1.png" width="700"></p>
</div>
<div id="qc-calculating-tmt-labeling-efficiency" class="section level2">
<h2 class="hasAnchor">
<a href="#qc-calculating-tmt-labeling-efficiency" class="anchor"></a>QC: Calculating TMT labeling efficiency</h2>
<p>TMT label efficiency can be measured as a proxy of peptides modified by TMT. This information is contained in the “evidence.txt” file, and is provided with this package:</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#Evidence data is provided with the package</span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="st">"evidence.raw"</span><span class="op">)</span>
<span class="va">evidencedata</span> <span class="op">&lt;-</span> <span class="va">evidence.raw</span></code></pre></div>
<p>To visualize TMT label efficiency, the plot_labelingefficiency function can be used.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"> <span class="fu"><a href="../reference/plot_labelingefficiency.html">plot_labelingefficiency</a></span><span class="op">(</span><span class="va">evidencedata</span><span class="op">)</span></code></pre></div>
<p><img src="protrusionproteome_BC_files/figure-html/plot-labelingefficiency-1.png" width="700"></p>
</div>
<div id="getting-aggregated-protein-data" class="section level2">
<h2 class="hasAnchor">
<a href="#getting-aggregated-protein-data" class="anchor"></a>Getting aggregated protein data</h2>
<p>Protein intensities are obtained from aggregated peptides over protein groups. This information is contained in the “proteinGroups.txt” file, is provided with this package and is used as input for the downstream analysis.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># The data is provided with the package</span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="st">"prot.raw"</span><span class="op">)</span>
<span class="va">data</span> <span class="op">&lt;-</span> <span class="va">prot.raw</span></code></pre></div>
<p>This dataset has the following dimensions:</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/dim.html">dim</a></span><span class="op">(</span><span class="va">data</span><span class="op">)</span></code></pre></div>
<pre><code>## [1] 5905  107</code></pre>
</div>
<div id="filtering-the-data" class="section level2">
<h2 class="hasAnchor">
<a href="#filtering-the-data" class="anchor"></a>Filtering the data</h2>
<p>We filter for decoy database hits, contaminant proteins and hits only identified by site, which are indicated by “+” in the columns “Reverse,”“Potential.contaminants” and “Only.identified.by.site,” respectively.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">proteins_filtered</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/filter_MaxQuant.html">filter_MaxQuant</a></span><span class="op">(</span><span class="va">data</span>,
    tofilter<span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Reverse"</span> , <span class="st">"Potential.contaminant"</span> ,<span class="st">"Only.identified.by.site"</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code>## Filtering based on 'Reverse', 'Potential.contaminant', 'Only.identified.by.site' column(s)</code></pre>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/dim.html">dim</a></span><span class="op">(</span><span class="va">proteins_filtered</span><span class="op">)</span></code></pre></div>
<pre><code>## [1] 5541  107</code></pre>
</div>
<div id="making-unique-identifiers" class="section level2">
<h2 class="hasAnchor">
<a href="#making-unique-identifiers" class="anchor"></a>Making unique identifiers</h2>
<p>Uniprot names are protein unique identifiers but are not immediately informative. The associated Gene names are informative, however these are not always unique (i.e Gene.names is not primary key to the proteins_filtered table).</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Are Gene.names primary key to this table?</span>
<span class="va">proteins_filtered</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_by.html">group_by</a></span><span class="op">(</span><span class="va">Gene.names</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/summarise.html">summarize</a></span><span class="op">(</span>count<span class="op">=</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/context.html">n</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> 
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/arrange.html">arrange</a></span><span class="op">(</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/desc.html">desc</a></span><span class="op">(</span> <span class="va">count</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span><span class="op">(</span> <span class="va">count</span> <span class="op">&gt;</span> <span class="fl">1</span><span class="op">)</span></code></pre></div>
<pre><code>## `summarise()` ungrouping output (override with `.groups` argument)</code></pre>
<pre><code>## # A tibble: 94 x 2
##    Gene.names count
##    &lt;chr&gt;      &lt;int&gt;
##  1 ""            18
##  2 "PLEC"         6
##  3 "CAST"         4
##  4 "ANXA2"        3
##  5 "HLA-C"        3
##  6 "LSM14B"       3
##  7 "NAP1L1"       3
##  8 "PCBP2"        3
##  9 "ABCE1"        2
## 10 "ABCF1"        2
## # … with 84 more rows</code></pre>
<p>Even more critical, some proteins do not have an annotated Gene name. Similarly to <span class="citation">(Xiaofei Zhang 2018)</span> approach, for those proteins missing a Gene name identifier we will use the Uniprot ID.</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">data_unique</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/make_unique.html">make_unique</a></span><span class="op">(</span><span class="va">proteins_filtered</span>, <span class="st">"Gene.names"</span>, <span class="st">"Protein.IDs"</span>, delim <span class="op">=</span> <span class="st">";"</span><span class="op">)</span></code></pre></div>
<p>We can check that name variable uniquely identifies proteins.</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">data_unique</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_by.html">group_by</a></span><span class="op">(</span><span class="va">name</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/count.html">tally</a></span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span><span class="op">(</span><span class="va">n</span> <span class="op">&gt;</span> <span class="fl">1</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/base/nrow.html">nrow</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code>## [1] 0</code></pre>
</div>
</div>
<div id="understanding-which-columns-will-be-used-in-the-summarizedexperiment-object" class="section level1">
<h1 class="hasAnchor">
<a href="#understanding-which-columns-will-be-used-in-the-summarizedexperiment-object" class="anchor"></a>Understanding which columns will be used in the SummarizedExperiment object</h1>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">columns_positions</span><span class="op">&lt;-</span><span class="fu">str_which</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">data_unique</span><span class="op">)</span>,<span class="st">"Reporter.intensity.corrected.(\\d)+.(\\d)"</span><span class="op">)</span>
<span class="va">intensities</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">data_unique</span><span class="op">)</span><span class="op">[</span><span class="fu">str_which</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">data_unique</span><span class="op">)</span>,<span class="st">"Reporter.intensity.corrected.(\\d)+.(\\d)"</span><span class="op">)</span><span class="op">]</span>
<span class="va">time_unit</span><span class="op">=</span><span class="fl">30</span>
<span class="va">time_span</span><span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">1</span>,<span class="fl">2</span>,<span class="fl">4</span>,<span class="fl">8</span>,<span class="fl">16</span><span class="op">)</span>
<span class="va">experiment</span> <span class="op">&lt;-</span> <span class="fu">str_c</span><span class="op">(</span><span class="fu"><a href="http://lubridate.tidyverse.org/reference/minute.html">minute</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/rep.html">rep</a></span><span class="op">(</span><span class="fu"><a href="http://lubridate.tidyverse.org/reference/period.html">minutes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">time_unit</span><span class="op">)</span> <span class="op">*</span><span class="va">time_span</span>,each<span class="op">=</span><span class="fl">2</span><span class="op">)</span><span class="op">)</span>,<span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"body"</span>,<span class="st">"prot"</span><span class="op">)</span>,sep<span class="op">=</span><span class="st">"_"</span><span class="op">)</span>
<span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html">kable</a></span><span class="op">(</span><span class="fu"><a href="https://tibble.tidyverse.org/reference/tibble.html">tibble</a></span><span class="op">(</span><span class="va">intensities</span>,<span class="va">experiment</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<table class="table">
<thead><tr class="header">
<th align="left">intensities</th>
<th align="left">experiment</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">Reporter.intensity.corrected.1.1</td>
<td align="left">30_body</td>
</tr>
<tr class="even">
<td align="left">Reporter.intensity.corrected.2.1</td>
<td align="left">30_prot</td>
</tr>
<tr class="odd">
<td align="left">Reporter.intensity.corrected.3.1</td>
<td align="left">60_body</td>
</tr>
<tr class="even">
<td align="left">Reporter.intensity.corrected.4.1</td>
<td align="left">60_prot</td>
</tr>
<tr class="odd">
<td align="left">Reporter.intensity.corrected.5.1</td>
<td align="left">120_body</td>
</tr>
<tr class="even">
<td align="left">Reporter.intensity.corrected.6.1</td>
<td align="left">120_prot</td>
</tr>
<tr class="odd">
<td align="left">Reporter.intensity.corrected.7.1</td>
<td align="left">240_body</td>
</tr>
<tr class="even">
<td align="left">Reporter.intensity.corrected.8.1</td>
<td align="left">240_prot</td>
</tr>
<tr class="odd">
<td align="left">Reporter.intensity.corrected.9.1</td>
<td align="left">480_body</td>
</tr>
<tr class="even">
<td align="left">Reporter.intensity.corrected.10.1</td>
<td align="left">480_prot</td>
</tr>
</tbody>
</table>
<div id="making-a-summarizedexperiment-object-from-tmt-data" class="section level2">
<h2 class="hasAnchor">
<a href="#making-a-summarizedexperiment-object-from-tmt-data" class="anchor"></a>Making a SummarizedExperiment object from TMT data</h2>
<p>SummarizedExperiment objects <span class="citation">(Martin Morgan 2020)</span> are widely used across Bioconductor packages as data containers. This class of object contains the actual data (assays), information on the samples (colData) and additional feature annotation. We generate the SummarizedExperiment object from our data extracting information directly from the column names of rectangular data using the <code>make_TMT_se</code> function. The actual assay data is log2-transformed of median-subtracted Prot/Cell-bodies for each condition.</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Generate a SummarizedExperiment object using using file information and user input</span>
<span class="va">columns_positions</span><span class="op">&lt;-</span><span class="fu">str_which</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">data_unique</span><span class="op">)</span>,<span class="st">"Reporter.intensity.corrected.(\\d)+.(\\d)"</span><span class="op">)</span>
<span class="va">se</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/make_TMT_se.html">make_TMT_se</a></span><span class="op">(</span><span class="va">data_unique</span>,<span class="va">columns_positions</span>,<span class="va">intensities</span>,time_unit<span class="op">=</span><span class="fl">30</span>,
time_span<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">1</span>,<span class="fl">2</span>,<span class="fl">4</span>,<span class="fl">8</span>,<span class="fl">16</span><span class="op">)</span>, numerator<span class="op">=</span> <span class="st">"prot"</span>, denominator<span class="op">=</span> <span class="st">"body"</span>, sep <span class="op">=</span> <span class="st">"_"</span><span class="op">)</span>

<span class="co"># Let's have a look at the SummarizedExperiment object</span>
<span class="va">se</span></code></pre></div>
<pre><code>## class: SummarizedExperiment 
## dim: 5541 5 
## metadata(0):
## assays(1): ''
## rownames(5541): RBM47 TTC26 ... MAU2 ENPP4
## rowData names(99): Protein.IDs Majority.protein.IDs ... name ID
## colnames(5): 30_1 60_1 120_1 240_1 480_1
## colData names(4): label ID condition replicate</code></pre>
<p>As we see the number of columns has been reduced to half (from 10 column sample data to 5 column sample data containing log2 Prot/Cell-bodies ratios)</p>
</div>
<div id="exploring-the-description-of-each-sample-in-the-summarizedexperiment-object" class="section level2">
<h2 class="hasAnchor">
<a href="#exploring-the-description-of-each-sample-in-the-summarizedexperiment-object" class="anchor"></a>Exploring the description of each sample in the SummarizedExperiment object</h2>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">SummarizedExperiment</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html">colData</a></span><span class="op">(</span><span class="va">se</span><span class="op">)</span></code></pre></div>
<pre><code>## DataFrame with 5 rows and 4 columns
##             label          ID   condition replicate
##       &lt;character&gt; &lt;character&gt; &lt;character&gt; &lt;numeric&gt;
## 30_1      30_prot        30_1          30         1
## 60_1      60_prot        60_1          60         1
## 120_1    120_prot       120_1         120         1
## 240_1    240_prot       240_1         240         1
## 480_1    480_prot       480_1         480         1</code></pre>
</div>
</div>
<div id="vizualing-the-results" class="section level1">
<h1 class="hasAnchor">
<a href="#vizualing-the-results" class="anchor"></a>Vizualing the results</h1>
<p>We can visualise the Prot/Cell-bodies distribution across timepoints.</p>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plot_scatter.html">plot_scatter</a></span><span class="op">(</span><span class="va">se</span>, <span class="fl">1</span>, <span class="fl">2</span>, <span class="st">"HIST"</span>, <span class="st">'orange'</span>, <span class="fl">4</span>, <span class="fl">4</span><span class="op">)</span></code></pre></div>
<p><img src="protrusionproteome_BC_files/figure-html/plot-scatter-1.png" width="700"> As we see Histone proteins are depleted from body.</p>
</div>
<div id="d-enrichment-analysis" class="section level1">
<h1 class="hasAnchor">
<a href="#d-enrichment-analysis" class="anchor"></a>1D enrichment analysis</h1>
<p>We can perform an protein enrichment analysis for a given timepoint</p>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">enrichment_table</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/enrich_1D.html">enrich_1D</a></span><span class="op">(</span><span class="va">se</span>,timepoint<span class="op">=</span> <span class="fl">5</span>,
                               dbs <span class="op">=</span> <span class="st">"GO_Molecular_Function_2018"</span>,
                              number_dbs<span class="op">=</span><span class="fl">1</span><span class="op">)</span></code></pre></div>
</div>
<div id="session-information" class="section level1">
<h1 class="hasAnchor">
<a href="#session-information" class="anchor"></a>Session information</h1>
<pre><code>## R version 3.6.3 (2020-02-29)
## Platform: x86_64-apple-darwin15.6.0 (64-bit)
## Running under: macOS Sierra 10.12.6
## 
## Matrix products: default
## BLAS:   /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRblas.0.dylib
## LAPACK: /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRlapack.dylib
## 
## locale:
## [1] en_GB.UTF-8/en_GB.UTF-8/en_GB.UTF-8/C/en_GB.UTF-8/en_GB.UTF-8
## 
## attached base packages:
## [1] parallel  stats4    stats     graphics  grDevices utils     datasets 
## [8] methods   base     
## 
## other attached packages:
##  [1] SummarizedExperiment_1.16.1 DelayedArray_0.12.3        
##  [3] BiocParallel_1.20.1         matrixStats_0.57.0         
##  [5] Biobase_2.46.0              GenomicRanges_1.38.0       
##  [7] GenomeInfoDb_1.22.1         IRanges_2.20.2             
##  [9] S4Vectors_0.24.4            BiocGenerics_0.32.0        
## [11] forcats_0.5.0               stringr_1.4.0              
## [13] purrr_0.3.4                 readr_1.4.0                
## [15] tidyr_1.1.2                 tibble_3.0.4               
## [17] tidyverse_1.3.0             lubridate_1.7.9.9001       
## [19] dplyr_1.0.2                 knitr_1.30                 
## [21] protrusionproteome_0.99.0   ggplot2_3.3.3              
## [23] BiocStyle_2.14.4           
## 
## loaded via a namespace (and not attached):
##  [1] bitops_1.0-6           fs_1.5.0               httr_1.4.2            
##  [4] rprojroot_2.0.2        tools_3.6.3            backports_1.2.1       
##  [7] utf8_1.1.4             R6_2.5.0               DBI_1.1.0             
## [10] colorspace_2.0-0       withr_2.3.0            tidyselect_1.1.0      
## [13] compiler_3.6.3         textshaping_0.2.1      cli_2.2.0             
## [16] rvest_0.3.6            enrichR_2.1            xml2_1.3.2            
## [19] desc_1.2.0             labeling_0.4.2         bookdown_0.21         
## [22] scales_1.1.1           pkgdown_1.6.1          systemfonts_0.3.2     
## [25] digest_0.6.27          rmarkdown_2.6          XVector_0.26.0        
## [28] pkgconfig_2.0.3        htmltools_0.5.1        highr_0.8             
## [31] dbplyr_2.0.0           rlang_0.4.10.9000      readxl_1.3.1          
## [34] rstudioapi_0.13        farver_2.0.3           generics_0.1.0        
## [37] jsonlite_1.7.2         RCurl_1.98-1.2         magrittr_2.0.1        
## [40] GenomeInfoDbData_1.2.2 Matrix_1.2-18          Rcpp_1.0.5            
## [43] munsell_0.5.0          fansi_0.4.1            lifecycle_0.2.0       
## [46] stringi_1.5.3          yaml_2.2.1             snakecase_0.11.0      
## [49] zlibbioc_1.32.0        grid_3.6.3             crayon_1.3.4          
## [52] lattice_0.20-38        haven_2.3.1            hms_0.5.3             
## [55] pillar_1.4.7           rjson_0.2.20           reprex_0.3.0          
## [58] glue_1.4.2             evaluate_0.14          BiocManager_1.30.10   
## [61] modelr_0.1.8           vctrs_0.3.6            cellranger_1.1.0      
## [64] gtable_0.3.0           assertthat_0.2.1       xfun_0.20             
## [67] janitor_2.0.1          broom_0.7.3            ragg_0.4.0            
## [70] memoise_1.1.0          ellipsis_0.3.1</code></pre>
</div>
<div id="references" class="section level1 unnumbered">
<h1 class="hasAnchor">
<a href="#references" class="anchor"></a>References</h1>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-Cox2014" class="csl-entry">
Jürgen Cox, Christian A Luber et al, Marco Hein. 2014. <span>“Accurate Proteome-Wide Label-Free Quantification by Delayed Normalization and Maximal Peptide Ratio Extraction, Termed MaxLFQ.”</span> <em>Mol Cell Proteomics</em>. <a href="https://doi.org/10.1074/mcp.M113.031591">https://doi.org/10.1074/mcp.M113.031591</a>.
</div>
<div id="ref-Dermit2020" class="csl-entry">
Maria Dermit, Flora Lee et al, Martin Dodel. 2020. <span>“Subcellular mRNA Localization Regulates Ribosome Biogenesis in Migrating Cells.”</span> <em>Developmental Cell</em> 55: 298–313.
</div>
<div id="ref-Morgan2020" class="csl-entry">
Martin Morgan, Jim Hester et al, Valerie Obenchain. 2020. <em>SummarizedExperiment: SummarizedExperiment Container</em>. <a href="https://bioconductor.org/packages/SummarizedExperiment">https://bioconductor.org/packages/SummarizedExperiment</a>.
</div>
<div id="ref-Arne2018" class="csl-entry">
Xiaofei Zhang, Gabrielle BA et al, Arne H Smits. 2018. <span>“Proteome-Wide Identification of Ubiquitin Interactions Using UbIA-MS.”</span> <em>Nature Protocols</em> 13: 530–50.
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Maria Dermit.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
